src/components/Icon/Icon.tsx

```diff
@@ -130,10 +136,26 @@ class Icon extends UIComponent<Extendable<IIconProps>, any> {
     )
   }
 
+  isFontBased(themeIcons: ThemeIcons): boolean {
+    const { name } = this.props
+    const themeIcon = themeIcons[name]
+
+    return !themeIcon || typeof themeIcon !== 'function'
+  }
+
+  getExtraProps({ icons }): IconExtraProps {
+    const isFontBased = this.isFontBased(icons)
+
+    return {
+      isFontBased,
+      fontIconFromTheme: isFontBased && icons[this.props.name],
+    }
+  }
+
   renderComponent({ ElementType, classes, rest, accessibility, icons }) {
```

Comment:
Propose we pass the entire `theme` to `renderComponent` as it is more generic and seems more useful in a wider range of scenarios.  Icons alone probably are not that useful to many other components.